File: bushop-src/src/types/scrape-job.ts
// src/types/scrape-job.ts
import type { ApiResponse } from './api';

export type ScrapeJobStatus = 'pending' | 'running' | 'completed' | 'failed';

export interface ScrapeJob {
  id: string;
  status: ScrapeJobStatus;
  route_id: number;
  created_at: string;
  completed_at: string | null;
}

export type ScrapeJobResponse = ApiResponse<ScrapeJob>;

export interface CreateScrapeJobRequest {
  route_id: number;
  travel_date: string;
}

File: bushop-scraper/app/schemas/scrape_job.py
from __future__ import annotations
import uuid
from datetime import datetime, date
from typing import Literal
from pydantic import BaseModel, ConfigDict

ScrapeJobStatus = Literal["pending", "running", "completed", "failed"]


class ScrapeJobCreate(BaseModel):
    route_id: int
    travel_date: date


class ScrapeJobResponse(BaseModel):
    id: uuid.UUID
    status: ScrapeJobStatus
    route_id: int
    created_at: datetime
    completed_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)
